### PR TITLE
Tokenization testing

### PR DESCRIPTION
**Pull request by sungshik**
Previously, only structural properties of generated TextMate grammars could conveniently be tested from within Rascal (e.g., expected number of rules). This PR adds the infra to test also behavioral properties (i.e., expected tokenization of example input text).

Specifically, the [`vscode-tmgrammar-test`](https://github.com/PanAeon/vscode-tmgrammar-test) tool is used to: (1) tokenize example input text using generated TextMate grammars; (2) compare actual tokenization against expected tokenization. The tool uses the same tokenization engine as VS Code.

CI is updated accordingly.

### CODE REVIEW

File: rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc [hunk 1]
```diff
@@ -158,7 +158,9 @@ default RegExp toRegExp(Grammar _, Condition c) {
     Converts a character range to a regular expression.
 }
 
-RegExp toRegExp(Grammar _, range(begin, end))
+RegExp toRegExp(Grammar _, range(char, char))
+    = regExp("<encode(char)>", []);
+default RegExp toRegExp(Grammar _, range(begin, end))
```
Comment (sungshik) [inline rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc:163]: These changes are unrelated to the goal of this PR and shouldn't be here, but they are...

File: vscode-extension/syntaxes/rascal.tmLanguage.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
```
Comment (sungshik) [inline vscode-extension/syntaxes/rascal.tmLanguage.json:1]: Changes in this file are due to the "extra" change in this PR

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -23,6 +23,10 @@ jobs:
           java-version: 11
           distribution: 'temurin'
           cache: 'maven'
+      - uses: actions/setup-node@v4
```
Comment (DavyLandman) [inline .github/workflows/build.yaml:26]: add which version of node we need. and enable caching to improve CI times.

File: rascal-textmate-core/.gitignore [hunk 4]
```diff
@@ -1 +1,6 @@
-target
\ No newline at end of file
+target
+
+# Only used to run tests:
+node_modules
+package-lock.json
+package.json
```
Comment (DavyLandman) [inline rascal-textmate-core/.gitignore:6]: Normally package.json is not in the .gitignore

File: rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc (hunk 1 above)
Comment (DavyLandman) [reply rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc:163]: tss ;-)

File: rascal-textmate-core/src/test/sh/lang/textmate/conversion-tests.bat [hunk 5]
```diff
@@ -0,0 +1,3 @@
+npx vscode-tmgrammar-test ^
```
Comment (DavyLandman) [inline rascal-textmate-core/src/test/sh/lang/textmate/conversion-tests.bat:1]: looks like the bat is the exact same.

how about we move it to `package.json` in the run block. and in rascal we just use `npm run conversion-tests` ?

File: vscode-extension/syntaxes/rascal.tmLanguage.json (hunk 2 above)
Comment (DavyLandman) [reply vscode-extension/syntaxes/rascal.tmLanguage.json:1]: ah, it's going to be a blast of a merge conflict with my other small PR. which might be a reason to pull those changes out of this PR and apply them on main after merging that?

**Review by DavyLandman — CHANGES_REQUESTED** [review]
LGTM, although I would try and use `package.json` to do some of the "command" stuff.

File: rascal-textmate-core/.gitignore (hunk 4 above)
Comment (sungshik) [reply rascal-textmate-core/.gitignore]: Fixed

File: vscode-extension/syntaxes/rascal.tmLanguage.json (hunk 2 above)
Comment (sungshik) [reply vscode-extension/syntaxes/rascal.tmLanguage.json:1]: Merged `main` into this branch, then resolved all conflicts in favor of `main`, then regenerated the file

File: rascal-textmate-core/src/test/sh/lang/textmate/conversion-tests.bat (hunk 5 above)
Comment (sungshik) [reply rascal-textmate-core/src/test/sh/lang/textmate/conversion-tests.bat]: Fixed:

  - Call `npx` directly from Rascal instead of via the scripts
  - Remove the scripts
